api/v1alpha3/awscluster_webhook.go

```diff
@@ -80,3 +82,25 @@ func (r *AWSCluster) ValidateUpdate(old runtime.Object) error {
 
 	return aggregateObjErrors(r.GroupVersionKind().GroupKind(), r.Name, allErrs)
 }
+
+func (r *AWSCluster) Default() {
+	// Default to Calico ingress rules if no rules have been set
+	if reflect.DeepEqual(r.Spec.NetworkSpec.CNI, CNISpec{}) {
```

Comment:
Once you make CNI a pointer, I think you should be able to check for nil.

I also think if we have a non-nil CNI field that has 0 IngressRules, we should **not** install the default Calico rules. WDYT @detiber @vincepri @randomvariable?

Reply:
Yes, that'd make sense to me.

Reply:
+1

Reply:
IngressRules is currently the only field for the CNISpec. Would the existence of a non nil CNI spec just mean that the cluster is misconfigured?

Reply:
Ignore my comment about non-nil with 0 rules (see other comment thread). If you have a non-nil CNI field, it means you are configuring at least 1 rule. So nil CNI = don't configure anything. non-nil CNI = configure at least 1 rule.